iOS_SDK/OneSignalSDK/Source/OneSignal.h

```diff
@@ -396,75 +410,69 @@ typedef NS_ENUM(NSUInteger, ONE_S_LOG_LEVEL) {
++ (void)sendTags:(NSDictionary* _Nonnull)keyValuePair onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)sendTags:(NSDictionary* _Nonnull)keyValuePair;
++ (void)sendTagsWithJsonString:(NSString* _Nonnull)jsonString;
++ (void)getTags:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)getTags:(OSResultSuccessBlock _Nullable)successBlock;
++ (void)deleteTag:(NSString* _Nonnull)key onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)deleteTag:(NSString* _Nonnull)key;
++ (void)deleteTags:(NSArray* _Nonnull)keys onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)deleteTags:(NSArray* _Nonnull)keys;
++ (void)deleteTagsWithJsonString:(NSString* _Nonnull)jsonString;
 
 + (OSPermissionSubscriptionState*)getPermissionSubscriptionState;
 
-+ (void)addPermissionObserver:(NSObject<OSPermissionObserver>*)observer;
-+ (void)removePermissionObserver:(NSObject<OSPermissionObserver>*)observer;
++ (void)addPermissionObserver:(NSObject<OSPermissionObserver>* _Nonnull)observer;
++ (void)removePermissionObserver:(NSObject<OSPermissionObserver>* _Nonnull)observer;
 
-+ (void)addSubscriptionObserver:(NSObject<OSSubscriptionObserver>*)observer;
-+ (void)removeSubscriptionObserver:(NSObject<OSSubscriptionObserver>*)observer;
++ (void)addSubscriptionObserver:(NSObject<OSSubscriptionObserver>* _Nonnull)observer;
++ (void)removeSubscriptionObserver:(NSObject<OSSubscriptionObserver>* _Nonnull)observer;
 
-+ (void)addEmailSubscriptionObserver:(NSObject<OSEmailSubscriptionObserver>*)observer;
-+ (void)removeEmailSubscriptionObserver:(NSObject<OSEmailSubscriptionObserver>*)observer;
++ (void)addEmailSubscriptionObserver:(NSObject<OSEmailSubscriptionObserver>* _Nonnull)observer;
++ (void)removeEmailSubscriptionObserver:(NSObject<OSEmailSubscriptionObserver>* _Nonnull)observer;
 
 + (void)setSubscription:(BOOL)enable;
 + (BOOL)isInAppMessagingPaused;
 + (void)pauseInAppMessages:(BOOL)pause;
 
 // - Posting Notification
-+ (void)postNotification:(NSDictionary*)jsonData;
-+ (void)postNotification:(NSDictionary*)jsonData onSuccess:(OSResultSuccessBlock)successBlock onFailure:(OSFailureBlock)failureBlock;
-+ (void)postNotificationWithJsonString:(NSString*)jsonData onSuccess:(OSResultSuccessBlock)successBlock onFailure:(OSFailureBlock)failureBlock;
++ (void)postNotification:(NSDictionary* _Nonnull)jsonData;
++ (void)postNotification:(NSDictionary* _Nonnull)jsonData onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)postNotificationWithJsonString:(NSString* _Nonnull)jsonData onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
 + (NSString*)parseNSErrorAsJsonString:(NSError*)error;
```

Comment:
`+ (NSString*)parseNSErrorAsJsonString:(NSError*)error;`
should have `_Nonnull` on return type and error?
`+ (NSString* _Nonnull)parseNSErrorAsJsonString:(NSError* _Nonnull)error;`